qemu/tests/ioeventfd.py

```diff
@@ -0,0 +1,166 @@
+                guest.
+            1.2 Dd file in the guest.
+                For rhel: dd if=/dev/zero of=/home/test bs=1M count=4096
+                oflag=direct.
+                For windows: We can load some big data to the disk via
+                DummyCMD.exe tools or other tools.
+            1.3 Reboot guest.
+            1.4 Iozone(Iozone.exe) -a in the guest.
+            1.5 Poweroff guest.
+            1.6 Repeat step 1.1-1.5 with ioeventfd=on.
+        Scenario 2:
+            2.1 Boot guest with ioeventfd=off.
+            2.2 Execute info qtree in QMP monitor, info qtree should show the
+            ioeventfd = false.
+            2.3 Check the ioeventfd=off via /proc/$PID/fd/.
+            2.4 Boot guest with ioeventfd=on.
+            2.5 Execute info qtree in QMP monitor, info qtree should show the
+            ioeventfd = true.
+            2.6 Check the ioeventfd=on via /proc/$PID/fd/.
+            2.7 Compare the output of 'ls -l /proc/$PID/fd/', the fds with
+            "off" should be less than the one with "on".
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _set_ioeventfd_options():
+        """ Set the ioeventfd options. """
+        if params['drive_format'] == 'virtio':
+            params['blk_extra_params_image1'] = ioeventfd
+            dev_id = 'image1'
+        elif params['drive_format'] == 'scsi-hd':
+            params['bus_extra_params_image1'] = ioeventfd
+            dev_id = 'virtio_scsi_pci0'
+        return dev_id
+
+    def _common_io_test():
+        """ Execute common io testing inside guest. """
+        logging.info('Doing io testing inside guest.')
+        op_cmd = params['op_cmd']
+        if os_type == 'windows':
+            op_cmd = utils_misc.set_winutils_letter(session, op_cmd)
+        o = session.cmd(op_cmd, float(params['op_timeout']))
+        logging.debug(o)
+
+    def _iozone_test(session):
+        """ Execute iozone testing inside guest. """
+        logging.info('Doing iozone inside guest.')
+        if os_type == 'windows':
+            if params['drive_format'] == 'scsi-hd':
+                session = utils_test.qemu.windrv_check_running_verifier(
+                    session, vm, test, params["driver_name"])
+        try:
+            iozone = generate_instance(params, session, 'iozone')
+            iozone.run(params['iozone_options'], float(params['iozone_timeout']))
+        finally:
+            iozone.clean()
+            return session
+
+    def _check_property(vm, ioeventfd_opt):
+        """
+        Check the value of ioeventfd by sending "info qtree" in QMP.
+        """
+        ioevent_qtree_val = 'true' if 'on' in ioeventfd_opt else 'false'
+        logging.info('Execute info qtree in QMP monitor.')
+        qtree = qemu_qtree.QtreeContainer()
+        qtree.parse_info_qtree(vm.monitor.human_monitor_cmd("info qtree"))
```

Comment:
Saw your step 2.2 Execute info qtree in QMP monitor, you here used hmp to check, better use qmp monitor to get info qtree, suggest use "qtree.parse_info_qtree(vm.monitor.info('qtree'))".

Reply:
ok, will update.